From 9baf13cf7d6a0298bd2f6782a88c1e145fa9b842 Mon Sep 17 00:00:00 2001
From: Bharath Ramesh <rameshb@vmware.com>
Date: Tue, 21 Oct 2014 17:25:57 -0700
Subject: [PATCH 5/7] Disable building IASyncOperation as part of shell
 extension

IAsyncOperation modifications are untested, disable building it.

Reviewed by: akreher, arolett
Review URL: https://reviewboard.eng.vmware.com/r/686986/

Conflicts:
	setup.py
---
 com/win32com/src/include/PythonCOMRegister.h |   2 +-
 com/win32comext/shell/src/shell.cpp          |   4 +-
 setup.py                                     | 229 ++++++++++++++++++---------
 3 files changed, 156 insertions(+), 79 deletions(-)

diff --git a/com/win32com/src/include/PythonCOMRegister.h b/com/win32com/src/include/PythonCOMRegister.h
index 95fe8b3..995e6fb 100644
--- a/com/win32com/src/include/PythonCOMRegister.h
+++ b/com/win32com/src/include/PythonCOMRegister.h
@@ -27,7 +27,7 @@ typedef struct
 #define PYCOM_INTERFACE_CLIENT_ONLY(ifc)	{ &IID_I##ifc, "I" #ifc, "IID_I" #ifc, &PyI##ifc::type, NULL }
 #define PYCOM_INTERFACE_SERVER_ONLY(ifc)	{ &IID_I##ifc, "I" #ifc, "IID_I" #ifc, NULL, GET_PYGATEWAY_CTOR(PyG##ifc)}
 #define PYCOM_INTERFACE_FULL(ifc)	{ &IID_I##ifc, "I" #ifc, "IID_I" #ifc, &PyI##ifc::type, GET_PYGATEWAY_CTOR(PyG##ifc)}
-#define PYCOM_INTERFACE_FULL_ASYNC(sdk_ifc,ifc)	{ &IID_I##sdk_ifc, "I" #ifc, "IID_I" #ifc, &PyI##ifc::type, GET_PYGATEWAY_CTOR(PyG##ifc)}
+//#define PYCOM_INTERFACE_FULL_ASYNC(sdk_ifc,ifc)	{ &IID_I##sdk_ifc, "I" #ifc, "IID_I" #ifc, &PyI##ifc::type, GET_PYGATEWAY_CTOR(PyG##ifc)}
 
 // Versions that use __uuidof() to get the IID, which seems to avoid the need
 // to link with a lib holding the IIDs.  Note that almost all extensions
diff --git a/com/win32comext/shell/src/shell.cpp b/com/win32comext/shell/src/shell.cpp
index 889353e..13f4316 100644
--- a/com/win32comext/shell/src/shell.cpp
+++ b/com/win32comext/shell/src/shell.cpp
@@ -46,7 +46,7 @@ generates Windows .hlp files.
 #include "PyIPersistFolder.h"
 #include "PyIColumnProvider.h"
 #include "PyIDropTargetHelper.h"
-#include "PyIAsyncOperation.h"
+//#include "PyIAsyncOperation.h"
 #include "PyIQueryAssociations.h"
 #include "PyIDockingWindow.h"
 #include "PyIDeskBand.h"
@@ -3582,7 +3582,7 @@ static const PyCom_InterfaceSupportInfo g_interfaceSupportData[] =
 	PYCOM_INTERFACE_IID_ONLY		  (ShellLinkW),
 	// rameshb: MS renamed AsyncOperation to DataObjectAsyncCapability
 	// This allows AsyncOperation interface to call DataObjectAsyncCapability
-	PYCOM_INTERFACE_FULL_ASYNC(DataObjectAsyncCapability, AsyncOperation),
+	//PYCOM_INTERFACE_FULL_ASYNC(DataObjectAsyncCapability, AsyncOperation),
 	PYCOM_INTERFACE_FULL(ContextMenu),
 	PYCOM_INTERFACE_SERVER_ONLY(ContextMenu2),
 	PYCOM_INTERFACE_SERVER_ONLY(ContextMenu3),
diff --git a/setup.py b/setup.py
index 368960b..83dcbe4 100644
--- a/setup.py
+++ b/setup.py
@@ -2020,84 +2020,161 @@ if get_build_version() < 11.0: # check if python was built with vs2012
                              sources=("""
                                       %(mapi)s/exchdapi.i         %(mapi)s/exchdapi.cpp
                                       """ % dirs).split()),
+    
+        WinExt_win32com('shell', libraries='shell32', pch_header="shell_pch.h",
+                        windows_h_version = 0x600,
+                        sources=("""
+                            %(shell)s/PyIActiveDesktop.cpp
+                            %(shell)s/PyIApplicationDestinations.cpp
+                            %(shell)s/PyIApplicationDocumentLists.cpp
+                            %(shell)s/PyIAsyncOperation.cpp
+                            %(shell)s/PyIBrowserFrameOptions.cpp
+                            %(shell)s/PyICategorizer.cpp
+                            %(shell)s/PyICategoryProvider.cpp
+                            %(shell)s/PyIColumnProvider.cpp
+                            %(shell)s/PyIContextMenu.cpp
+                            %(shell)s/PyIContextMenu2.cpp
+                            %(shell)s/PyIContextMenu3.cpp
+                            %(shell)s/PyICopyHook.cpp
+                            %(shell)s/PyICurrentItem.cpp
+                            %(shell)s/PyICustomDestinationList.cpp
+                            %(shell)s/PyIDefaultExtractIconInit.cpp
+                            %(shell)s/PyIDeskBand.cpp
+                            %(shell)s/PyIDisplayItem.cpp
+                            %(shell)s/PyIDockingWindow.cpp
+                            %(shell)s/PyIDropTargetHelper.cpp
+                            %(shell)s/PyIEnumExplorerCommand.cpp
+                            %(shell)s/PyIEnumIDList.cpp
+                            %(shell)s/PyIEnumObjects.cpp
+                            %(shell)s/PyIEnumResources.cpp
+                            %(shell)s/PyIEnumShellItems.cpp
+                            %(shell)s/PyIEmptyVolumeCache.cpp
+                            %(shell)s/PyIEmptyVolumeCacheCallBack.cpp
+                            %(shell)s/PyIExplorerBrowser.cpp
+                            %(shell)s/PyIExplorerBrowserEvents.cpp
+                            %(shell)s/PyIExplorerCommand.cpp
+                            %(shell)s/PyIExplorerCommandProvider.cpp
+                            %(shell)s/PyIExplorerPaneVisibility.cpp
+                            %(shell)s/PyIExtractIcon.cpp
+                            %(shell)s/PyIExtractIconW.cpp
+                            %(shell)s/PyIExtractImage.cpp
+                            %(shell)s/PyIFileOperation.cpp
+                            %(shell)s/PyIFileOperationProgressSink.cpp
+                            %(shell)s/PyIIdentityName.cpp
+                            %(shell)s/PyIInputObject.cpp
+                            %(shell)s/PyIKnownFolder.cpp
+                            %(shell)s/PyIKnownFolderManager.cpp
+                            %(shell)s/PyINameSpaceTreeControl.cpp
+                            %(shell)s/PyIObjectArray.cpp
+                            %(shell)s/PyIObjectCollection.cpp
+                            %(shell)s/PyIPersistFolder.cpp
+                            %(shell)s/PyIPersistFolder2.cpp
+                            %(shell)s/PyIQueryAssociations.cpp
+                            %(shell)s/PyIRelatedItem.cpp
+                            %(shell)s/PyIShellBrowser.cpp
+                            %(shell)s/PyIShellExtInit.cpp
+                            %(shell)s/PyIShellFolder.cpp
+                            %(shell)s/PyIShellFolder2.cpp
+                            %(shell)s/PyIShellIcon.cpp
+                            %(shell)s/PyIShellIconOverlay.cpp
+                            %(shell)s/PyIShellIconOverlayIdentifier.cpp
+                            %(shell)s/PyIShellIconOverlayManager.cpp
+                            %(shell)s/PyIShellItem.cpp
+                            %(shell)s/PyIShellItem2.cpp
+                            %(shell)s/PyIShellItemArray.cpp
+                            %(shell)s/PyIShellItemResources.cpp
+                            %(shell)s/PyIShellLibrary.cpp
+                            %(shell)s/PyIShellLink.cpp
+                            %(shell)s/PyIShellLinkDataList.cpp
+                            %(shell)s/PyIShellView.cpp
+                            %(shell)s/PyITaskbarList.cpp
+                            %(shell)s/PyITransferAdviseSink.cpp
+                            %(shell)s/PyITransferDestination.cpp
+                            %(shell)s/PyITransferMediumItem.cpp
+                            %(shell)s/PyITransferSource.cpp
+                            %(shell)s/PyIUniformResourceLocator.cpp
+                            %(shell)s/shell.cpp
+
+                            """ % dirs).split())
+    ]
+else:
+    com_extensions += [
+        WinExt_win32com('shell', libraries='shell32', pch_header="shell_pch.h",
+                        windows_h_version = 0x600,
+                        sources=("""
+                            %(shell)s/PyIActiveDesktop.cpp
+                            %(shell)s/PyIApplicationDestinations.cpp
+                            %(shell)s/PyIApplicationDocumentLists.cpp
+                            %(shell)s/PyIBrowserFrameOptions.cpp
+                            %(shell)s/PyICategorizer.cpp
+                            %(shell)s/PyICategoryProvider.cpp
+                            %(shell)s/PyIColumnProvider.cpp
+                            %(shell)s/PyIContextMenu.cpp
+                            %(shell)s/PyIContextMenu2.cpp
+                            %(shell)s/PyIContextMenu3.cpp
+                            %(shell)s/PyICopyHook.cpp
+                            %(shell)s/PyICurrentItem.cpp
+                            %(shell)s/PyICustomDestinationList.cpp
+                            %(shell)s/PyIDefaultExtractIconInit.cpp
+                            %(shell)s/PyIDeskBand.cpp
+                            %(shell)s/PyIDisplayItem.cpp
+                            %(shell)s/PyIDockingWindow.cpp
+                            %(shell)s/PyIDropTargetHelper.cpp
+                            %(shell)s/PyIEnumExplorerCommand.cpp
+                            %(shell)s/PyIEnumIDList.cpp
+                            %(shell)s/PyIEnumObjects.cpp
+                            %(shell)s/PyIEnumResources.cpp
+                            %(shell)s/PyIEnumShellItems.cpp
+                            %(shell)s/PyIEmptyVolumeCache.cpp
+                            %(shell)s/PyIEmptyVolumeCacheCallBack.cpp
+                            %(shell)s/PyIExplorerBrowser.cpp
+                            %(shell)s/PyIExplorerBrowserEvents.cpp
+                            %(shell)s/PyIExplorerCommand.cpp
+                            %(shell)s/PyIExplorerCommandProvider.cpp
+                            %(shell)s/PyIExplorerPaneVisibility.cpp
+                            %(shell)s/PyIExtractIcon.cpp
+                            %(shell)s/PyIExtractIconW.cpp
+                            %(shell)s/PyIExtractImage.cpp
+                            %(shell)s/PyIFileOperation.cpp
+                            %(shell)s/PyIFileOperationProgressSink.cpp
+                            %(shell)s/PyIIdentityName.cpp
+                            %(shell)s/PyIInputObject.cpp
+                            %(shell)s/PyIKnownFolder.cpp
+                            %(shell)s/PyIKnownFolderManager.cpp
+                            %(shell)s/PyINameSpaceTreeControl.cpp
+                            %(shell)s/PyIObjectArray.cpp
+                            %(shell)s/PyIObjectCollection.cpp
+                            %(shell)s/PyIPersistFolder.cpp
+                            %(shell)s/PyIPersistFolder2.cpp
+                            %(shell)s/PyIQueryAssociations.cpp
+                            %(shell)s/PyIRelatedItem.cpp
+                            %(shell)s/PyIShellBrowser.cpp
+                            %(shell)s/PyIShellExtInit.cpp
+                            %(shell)s/PyIShellFolder.cpp
+                            %(shell)s/PyIShellFolder2.cpp
+                            %(shell)s/PyIShellIcon.cpp
+                            %(shell)s/PyIShellIconOverlay.cpp
+                            %(shell)s/PyIShellIconOverlayIdentifier.cpp
+                            %(shell)s/PyIShellIconOverlayManager.cpp
+                            %(shell)s/PyIShellItem.cpp
+                            %(shell)s/PyIShellItem2.cpp
+                            %(shell)s/PyIShellItemArray.cpp
+                            %(shell)s/PyIShellItemResources.cpp
+                            %(shell)s/PyIShellLibrary.cpp
+                            %(shell)s/PyIShellLink.cpp
+                            %(shell)s/PyIShellLinkDataList.cpp
+                            %(shell)s/PyIShellView.cpp
+                            %(shell)s/PyITaskbarList.cpp
+                            %(shell)s/PyITransferAdviseSink.cpp
+                            %(shell)s/PyITransferDestination.cpp
+                            %(shell)s/PyITransferMediumItem.cpp
+                            %(shell)s/PyITransferSource.cpp
+                            %(shell)s/PyIUniformResourceLocator.cpp
+                            %(shell)s/shell.cpp
+
+                            """ % dirs).split())
     ]
 com_extensions += [
-    WinExt_win32com('shell', libraries='shell32', pch_header="shell_pch.h",
-                    windows_h_version = 0x600,
-                    sources=("""
-                        %(shell)s/PyIActiveDesktop.cpp
-                        %(shell)s/PyIApplicationDestinations.cpp
-                        %(shell)s/PyIApplicationDocumentLists.cpp
-                        %(shell)s/PyIAsyncOperation.cpp
-                        %(shell)s/PyIBrowserFrameOptions.cpp
-                        %(shell)s/PyICategorizer.cpp
-                        %(shell)s/PyICategoryProvider.cpp
-                        %(shell)s/PyIColumnProvider.cpp
-                        %(shell)s/PyIContextMenu.cpp
-                        %(shell)s/PyIContextMenu2.cpp
-                        %(shell)s/PyIContextMenu3.cpp
-                        %(shell)s/PyICopyHook.cpp
-                        %(shell)s/PyICurrentItem.cpp
-                        %(shell)s/PyICustomDestinationList.cpp
-                        %(shell)s/PyIDefaultExtractIconInit.cpp
-                        %(shell)s/PyIDeskBand.cpp
-                        %(shell)s/PyIDisplayItem.cpp
-                        %(shell)s/PyIDockingWindow.cpp
-                        %(shell)s/PyIDropTargetHelper.cpp
-                        %(shell)s/PyIEnumExplorerCommand.cpp
-                        %(shell)s/PyIEnumIDList.cpp
-                        %(shell)s/PyIEnumObjects.cpp
-                        %(shell)s/PyIEnumResources.cpp
-                        %(shell)s/PyIEnumShellItems.cpp
-                        %(shell)s/PyIEmptyVolumeCache.cpp
-                        %(shell)s/PyIEmptyVolumeCacheCallBack.cpp
-                        %(shell)s/PyIExplorerBrowser.cpp
-                        %(shell)s/PyIExplorerBrowserEvents.cpp
-                        %(shell)s/PyIExplorerCommand.cpp
-                        %(shell)s/PyIExplorerCommandProvider.cpp
-                        %(shell)s/PyIExplorerPaneVisibility.cpp
-                        %(shell)s/PyIExtractIcon.cpp
-                        %(shell)s/PyIExtractIconW.cpp
-                        %(shell)s/PyIExtractImage.cpp
-                        %(shell)s/PyIFileOperation.cpp
-                        %(shell)s/PyIFileOperationProgressSink.cpp
-                        %(shell)s/PyIIdentityName.cpp
-                        %(shell)s/PyIInputObject.cpp
-                        %(shell)s/PyIKnownFolder.cpp
-                        %(shell)s/PyIKnownFolderManager.cpp
-                        %(shell)s/PyINameSpaceTreeControl.cpp
-                        %(shell)s/PyIObjectArray.cpp
-                        %(shell)s/PyIObjectCollection.cpp
-                        %(shell)s/PyIPersistFolder.cpp
-                        %(shell)s/PyIPersistFolder2.cpp
-                        %(shell)s/PyIQueryAssociations.cpp
-                        %(shell)s/PyIRelatedItem.cpp
-                        %(shell)s/PyIShellBrowser.cpp
-                        %(shell)s/PyIShellExtInit.cpp
-                        %(shell)s/PyIShellFolder.cpp
-                        %(shell)s/PyIShellFolder2.cpp
-                        %(shell)s/PyIShellIcon.cpp
-                        %(shell)s/PyIShellIconOverlay.cpp
-                        %(shell)s/PyIShellIconOverlayIdentifier.cpp
-                        %(shell)s/PyIShellIconOverlayManager.cpp
-                        %(shell)s/PyIShellItem.cpp
-                        %(shell)s/PyIShellItem2.cpp
-                        %(shell)s/PyIShellItemArray.cpp
-                        %(shell)s/PyIShellItemResources.cpp
-                        %(shell)s/PyIShellLibrary.cpp
-                        %(shell)s/PyIShellLink.cpp
-                        %(shell)s/PyIShellLinkDataList.cpp
-                        %(shell)s/PyIShellView.cpp
-                        %(shell)s/PyITaskbarList.cpp
-                        %(shell)s/PyITransferAdviseSink.cpp
-                        %(shell)s/PyITransferDestination.cpp
-                        %(shell)s/PyITransferMediumItem.cpp
-                        %(shell)s/PyITransferSource.cpp
-                        %(shell)s/PyIUniformResourceLocator.cpp
-                        %(shell)s/shell.cpp
-
-                        """ % dirs).split()),
-
     WinExt_win32com('propsys', libraries='propsys', delay_load_libraries='shell32',
                     unicode_mode=True,
                     sources=("""
-- 
2.8.0.windows.1

